Exec/test_problems/imposed_external_heating/MaestroMakeS.cpp

```diff
@@ -0,0 +1,573 @@
+
+                    if (use_delta_gamma1_term_loc) {
+                        delta_gamma1_arr(i, j, k) =
+                            eos_state.gam1 - gamma1bar_arr(i, j, k);
+
+                        Real U_dot_er = 0.0;
+                        for (auto n = 0; n < AMREX_SPACEDIM; ++n) {
+                            U_dot_er +=
+                                u_arr(i, j, k, n) * normal_arr(i, j, k, n);
+                        }
+
+                        delta_gamma1_term_arr(i, j, k) =
+                            (eos_state.gam1 - gamma1bar_arr(i, j, k)) *
+                            u_arr(i, j, k, AMREX_SPACEDIM - 1) *
+                            gradp0_cart_arr(i, j, k) /
+                            (gamma1bar_arr(i, j, k) * gamma1bar_arr(i, j, k) *
+                             p0_arr(i, j, k));
+                    } else {
+                        delta_gamma1_term_arr(i, j, k) = 0.0;
+                        delta_gamma1_arr(i, j, k) = 0.0;
+                    }
+                });
+#endif
+            } else {
+                const auto anelastic_cutoff_density_coord_lev =
+                    base_geom.anelastic_cutoff_density_coord(lev);
+
+                AMREX_PARALLEL_FOR_3D(tileBox, i, j, k, {
+                    eos_t eos_state;
+
+                    eos_state.rho = scal_arr(i, j, k, Rho);
+                    eos_state.T = scal_arr(i, j, k, Temp);
+                    for (auto n = 0; n < NumSpec; ++n) {
+                        eos_state.xn[n] =
+                            scal_arr(i, j, k, FirstSpec + n) / eos_state.rho;
+                    }
+#if NAUX_NET > 0
+                    for (auto n = 0; n < NumAux; ++n) {
+                        eos_state.aux[n] =
+                            scal_arr(i, j, k, FirstAux + n) / eos_state.rho;
+                    }
+#endif
+
+                    // dens, temp, and xmass are inputs
+                    eos(eos_input_rt, eos_state);
+
+                    auto eos_xderivs = composition_derivatives(eos_state);
+
+                    Real sigma =
+                        eos_state.dpdT /
+                        (eos_state.rho * eos_state.cp * eos_state.dpdr);
+
+                    Real xi_term = 0.0;
+                    Real pres_term = 0.0;
+
+                    if (use_omegadot_terms_in_S_loc) {
+                        for (auto comp = 0; comp < NumSpec; ++comp) {
+                            xi_term -= eos_xderivs.dhdX[comp] *
+                                       rho_odot_arr(i, j, k, comp) /
+                                       eos_state.rho;
+
+                            pres_term += eos_xderivs.dpdX[comp] *
+                                         rho_odot_arr(i, j, k, comp) /
+                                         eos_state.rho;
+                        }
+                    }
+
+                    S_cc_arr(i, j, k) =
+                        //(sigma / eos_state.rho) *
+                        (1.0 / eos_state.rho) *
```

Comment:
This is the only difference between the MaestroMakeS.cpp in MAESTROeX/Source.  Can you explain why you want this difference?

Reply:
Yes, this is unfortunate. It should ideally be removed. I want to set sigma=1 in this test, or otherwise adapt the form of the heating. Perhaps I can factor it into `make_heating.F90` (I've not checked to see what is available to Fortran yet), unless there is a better suggestion.

Reply:
What's the physical explanation for wanting sigma=1?

Reply:
Other than having some kind of density and pressure gradient, there is not much physical about this test. Simply we know how the exact solution should evolve. I just want to implement a specific form of heating so that dw0/dz=S. sigma is not important for this test -- I just want to avoid having to calculate it, but if it is available it could be factored into rho_Hext in make_heating. I'll look to do that unless you can think of something more elegant.

Reply:
If it's only needed for this test problem and not a general feature we need for a larger class of problems, this one-off local file seems ok.